src/bt-gatt-io.c

```diff
@@ -307,17 +307,17 @@ client_register_client_cb(int status, int client_if, bt_uuid_t* app_uuid)
 static void
 client_scan_result_cb(bt_bdaddr_t* bd_addr, int rssi, uint8_t* adv_data)
 {
-  size_t len;
+  size_t len = 0;
```

Comment:
Nit: please put the initializations at the end of the declarations block. Improves compatibility with the existing C dialects.